app/modules/video-players/jwplayer-video-ads.js

```diff
@@ -10,23 +10,35 @@ export default class JWPlayerVideoAds {
 		if (this.params.noAds) {
 			return Promise.resolve(this.params);
 		} else if (this.isA9VideoEnabled()) {
-			return this.parseBidderParameters()
-				/* eslint no-console: 0 */
-				.catch((error) => console.error('JWPlayer: Error while receiving bidder parameters', error));
+			return new Promise((resolve) => {
+				this.parseBidderParameters(resolve, (params) => {
+					/* eslint no-console: 0 */
+					console.error('JWPlayer: Error while receiving bidder parameters');
+					resolve(params);
+				});
+			});
 		} else {
 			return Promise.resolve({});
 		}
 	}
 
-	parseBidderParameters() {
+	parseBidderParameters(onSuccess, onError) {
 		const a9 = Ads.getInstance().a9;
+		const responseTimeout = 2000;
 
 		if (!a9) {
-			return Promise.resolve({});
+			onError({});
 		}
 
-		return a9.waitForResponse()
-			.then(() => a9.getSlotParams('FEATURED'));
+		a9.waitForResponse(
```

Comment:
I think we need more here and be ready to handle to versions of `a9.waitForResponse()`. What do you think of throwing an exception in the new version when it doesn't get callbacks, catching it here and then fallback to the old "promise version"?

Reply:
Done